### PR TITLE
chore: prepare tests to expire feature flag 'kms:defaultKeyPolicies'

### PR DESCRIPTION
**Pull request by nija-at**
This is a follow up to - 31b1b32.

Unit tests in the `aws-ec2` and `aws-s3-notifications` modules
assert resources generated by KMS.

The '@aws-cdk/aws-kms:defaultKeyPolicies' feature flag will be
reset in CDKv2. Update the necessary tests so that they continue
to work when this happens.

Two course of actions are taken here - some tests needed to
be made more 'precise' in their assertions, and others had to be
configured to only run in CDKv1.

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk/aws-dynamodb/test/dynamodb.test.ts [hunk 1]
```diff
@@ -486,154 +487,10 @@ test('fails if both replication regions used with customer managed CMK', () => {
   })).toThrow('TableEncryption.CUSTOMER_MANAGED is not supported by DynamoDB Global Tables (where replicationRegions was set)');
 });
 
-test('if an encryption key is included, decrypt permissions are also added for grantStream', () => {
```
Comment (nija-at) [inline packages/@aws-cdk/aws-dynamodb/test/dynamodb.test.ts:489]: Not sure what this test is trying to verify. Dropping it entirely.

Comment (nija-at) [reply packages/@aws-cdk/aws-dynamodb/test/dynamodb.test.ts:489]: But the assertion does not actually verify decrypt permission on the user and the associated policy, does it?

Comment (nija-at) [reply packages/@aws-cdk/aws-dynamodb/test/dynamodb.test.ts:489]: Further analysis -

[`grantStreamRead()`](https://github.com/aws/aws-cdk/blob/a0bdffc423f25af37fd6ce3970167f947fd027a0/packages/%40aws-cdk/aws-dynamodb/lib/table.ts#L654-L656) calls `combinedGrant()` API without any `tableActions` configured.

This means that the key permissions [will not be updated](https://github.com/aws/aws-cdk/blob/a0bdffc423f25af37fd6ce3970167f947fd027a0/packages/%40aws-cdk/aws-dynamodb/lib/table.ts#L899-L903).

Seems right to drop this test. I'm guessing reading a stream does not require permissions to the key, only reading the table does.

Comment (njlynch) [reply packages/@aws-cdk/aws-dynamodb/test/dynamodb.test.ts:489]: Huh, you're right. This test was never testing anything. Weird...

**Review by njlynch — APPROVED** [review]

**Pull request merged**